### ISSUE TITLE
Add option for visual studio debugging

### ISSUE THREAD

**Issue by michaelcheers** [bridgedotnet/Bridge#1308]
I was thinking it would be cool if you could debug the Bridge Code in Visual Studio using the native debugger. That way you could debug the C# code.

**Comment by geoffreymcgill** [comment]
Yes, agreed. This functionality is a high priority for us.

**Comment by michaelcheers** [comment]
See [the MSDN page](https://msdn.microsoft.com/en-us/library/bb145934.aspx) about it.

**Comment by geoffreymcgill** [comment]
Related to #259.

**Comment by alex4998** [comment]
I'm highly interested in this feature and wonder if there is a progress in this direction. I see the feature was requested and sort of accepted more than half a year ago but how much longer it will take until it is implemented? I would like to use .NET Bridge in our next project but I don't think it will be possible without this feature.

**Comment by geoffreymcgill** [comment]
>  but I don't think it will be possible without this feature.

Can you provide some insight into why [lacking] this feature would prevent Bridge from being used on your project? genuinely interested

**Comment by alex4998** [comment]
1. Most of the people in our team are not proficient in javascript and frankly speaking are not interested in learning its details which is totally fine in our case. javascript was not required in our development process before.
2. Personal preference of all of them is being able to write and debug the same code source without doing mental exercising of mapping javascript to what was written in C# no matter how similar the code looks. From our experience writing the code is about 30% or less of the time but debugging is 70% or more. Thus, most of the time you'll be forced to look at and read javascript.

Then the most important point is that we strongly believe that using javascript directly won't allow us to efficiently implement the complex algorithms we are required to implement. With all above said, for us it is a clear choice either to use the TypeScript, going through a minor learning curve or wait until .NET Bridge source map support is implemented and use C# in all our work. The question is if the waiting time for .NET Bridge is reasonable.

**Comment by Zaid-Ajaj** [comment]
@alex4998, debugging the generated javascript with either chrome or firefox (with Firebug) gives you a very similar experience to the one visual studio offers. The javascript is extremely readable. 

Knowing a bit of javascript is crucial to any web developement you might want to do especially if you are talking about implementing complex algorithms. Because even if it "just works", you do want to look at the generated code to do **optimizations**. 

TL;DR: Using Bridge.NET without knowing Javascript is like using Entity Framework without knowing any SQL.

**Comment by TFTomSun** [comment]
3 years later ... any progress on visual studio debugging? :)

**Comment by TFTomSun** [comment]
btw. i discovered that visual studio debugging is already possible.
- copy the c# sources next to the JavaScript scources
- enable source map
- set breakpoint in copied c# file 
- start chrome with debugger port enabled
- attach visual studio with JavaScript (chrome) debug engine

I'm pretty sure that can be automated as a one click solution.
